pyccel/codegen/printing/fcode.py

```diff
@@ -657,13 +657,14 @@ def _print_NumpyEmpty(self, expr):
 
     def _print_NumpyNorm(self, expr):
         """Fortran print."""
-
-        if expr.dim:
-            rhs = 'Norm2({},{})'.format(self._print(expr.arg),self._print(expr.dim))
+        arg = PythonAbs(expr.arg) if isinstance(expr.arg.dtype, NativeComplex) else expr.arg
+        if expr.axis:
+            axis = str(abs(expr.axis.python_value - arg.rank)) + '_C_DOUBLE'
```

Comment:
Please use pyccel nodes rather than python values, this way we will be sure that the kind tag is correct and it can be easily modified everywhere if we need to change it (e.g. when we fix #604 )